README.md
### KeepInTouch

A Customer Relationship Management (CRM) tool for the Content Management System [WebsiteBaker] [1] and [LEPTON CMS] [2]

This tool is designed as Admin-Tool an can be accessed and extended by other Add-ons. A lot of **kitXXX** extensions are already available, please check [phpManufaktur at GitHub] [3]. 

#### Requirements

* minimum PHP 5.2.x
* using [WebsiteBaker] [1] _or_ using [LEPTON CMS] [2]
* [dbConnect_LE] [4] must be installed
* [Dwoo] [5] must be installed 

#### Installation

* download the actual [KeepInTouch_x.xx.zip] [6] installation archive
* in CMS backend select the file from "Add-ons" -> "Modules" -> "Install module"

#### First Steps

Please visit the [phpManufaktur] [7] to get more informations about **KeepInTouch**.

Use the [phpManufaktur - General Addons Support Group] [8] to get support.

[1]: http://websitebaker2.org "WebsiteBaker Content Management System"
[2]: http://lepton-cms.org "LEPTON CMS"
[3]: https://github.com/phpManufaktur
[4]: https://github.com/phpManufaktur/dbConnect_LE/downloads
[5]: https://github.com/phpManufaktur/Dwoo/downloads
[6]: https://github.com/phpManufaktur/KeepInTouch/downloads
[7]: https://phpmanufaktur.de
[8]: https://phpmanufaktur.de/support


CHANGELOG
### KeepInTouch - Customer Relationship Management (CRM)
(c) 2010 - 2012 by phpManufaktur
Ralf Hertsch, Stockholmer Str. 31, 13359 Berlin (Germany)
http://phpManufaktur.de - ralf.hertsch@phpManufaktur.de

Release 0.75 - 2013-11-22

* fixed missing parts: problem accessing language files at BlackCat CMS environment
* added icon.png (BlackCat CMS) and tool_icon.png (WebsiteBaker 2.8.4)

Release 0.74 - 2013-11-21

* fixed: problem accessing language files at BlackCat CMS environment
* changed: precheck.php does no longer check for Dwoo at BlackCat CMS environment

Release 0.73 - 2013-09-24

* fixed: it was not possible to import 2-column massmail CSV files because KIT
  expected at minimum 3 columns.

Release 0.72 - 2013-05-23

* fixed: WebsiteBaker does not like the parameter 'ctab' at provider configuration

Release 0.71 - 2013-05-06

* added: admin function to delete contacts of a selected category

Release 0.70 - 2013-03-20

* added: fields kit_identifier and kit_contact_since are now accessible for
  the import and for kitForm
* changed: import accept now category values or identifiers
* added: import can activate newsletter automatically
* added: truncate KIT contact database before import new data
* added: import can handle internal categories
* added: import ignore records with status deleted
* added: switch to abort the actual running newsletter cronjob

Release 0.69 - 2013-01-21

* added: additional address fields EXTRA and REGION
* fixed: problem with import of distributions, newsletters and free fields

Release 0.68 - 2013-01-14

* added: KIT interface now handles KIT categories of type DISTRIBUTION
* changed: kitContactInterface::getCategories() added parameter $use_keys return
  a structured categories array
* added: kitContactInterface::getContactsByCategory return all KIT_ID's of the
  members of the specified KIT category

Release 0.67 - 2013-01-08

* fixed: KIT interface does not add the country to the contact array
* fixed: install, uninstall and upgrade try to include the not existing class.import.php

Release 0.66 - 2013-01-04

* added export dialog, please read <http://yals.de/WpBd3v> for more information

Release 0.65 - 2013-01-02

* added import dialog, please read <http://yals.de/VAu8gI> for more information

Release 0.64 - 2012-12-08

* added field kit_contact_language to the KIT interface output array
* added KIT special functions to enable the contacts to upload and download
  files directly to and from the protected directories below /media/kit_protected

Release 0.63 - 2012-10-18

* fixed a problem getting a not existing email address in class.interface.php

Release 0.62 - 2012-10-05

* added ability to change the table prefix easy via config.json, please read
  <http://yals.de/OQRv0Y> for more information
* added free fields and free notes to the contact array returned by the
  class.interface.php
* changed implementation of the Dwoo Template Engine
* changed contact list: add any field in any order, sort, filter and search
  all contacts, copy to clipboard, export as CSV or tabbed CSV fo Excel, create
  a PDF or print the table.

Release 0.61 - 2012-08-08

* added optional parameter $get_deleted to getContact() in kitContactInterface
  to enable getting also informations about deleted contacts

Release 0.60 - 2012-07-27

* changed WebsiteBaker version check
* fixed a problem in newsletter processing - the error handling of KIT was too
  strict and caused interrupts and a loop in the package sending.

Release 0.59 - 2012-06-13

* Changed order of initialisation in class.newsletter.cfg.php, class.config.php,
  class.newsletter.php and class.contact.php (PHP 5.3.x compatibility)
* marked showOSMmap() as deprecated, Google Maps v2 are no longer supported
* changed extending of class PHPMailer for usage with WebsiteBaker 2.8.3
* added some additional checks and remarks

Release 0.58 - 2012-06-07

* fixed a problem with handling of special chars while sending out newsletters
  and distributions (cronjob.php)
* reformatted CHANGELOG, fixed some typos

Release 0.57 - 2012-06-06

* moved repository from SVN to GitHub
* added CHANGELOG
* changed license from GPL to MIT
* added LICENSE
* added README.md
* deleted info.txt
* changed all header and archive informations

Release 0.56 - 2012-05-09

* fixed - uninstall calls the no longer existing class.dialogs.php
* added - LEPTON I18n language support
* moved - the old static languages files are renamed to old.* and still in use,
  they will be obsolete step by step with the next KIT releases

Release 0.55 - 2012-02-29

* added - KIT option cfgNewsletterAccountInfo enables to switch on/off the
  account information for the user.

Release 0.54 - 2012-02-19

* added - language table (ISO 639-1) for prefered languages
* added - field "contact language" in KIT contact dialog
* added - field "kit_contact_language" in KIT interface
* added - KIT options cfgContactLanguageDefault and cfgContactLanguageSelect
* added - KIT option cfgNewsletterLanguageMarkers enables to define language
  depending newsletters

Release 0.53 - 2012-02-14

* removed - frontend.css is no longer needed
* changed - introduce $_SESSION['KIT_HTML_REQUEST'] to mark fields which should
  not processed by xssPrevent()
* added - check the language files for correct UTF-8 encoding

Release 0.52 - 2012-02-04

* added - Country support for the KIT interface
* fixed - problem inserting and updating addresses, KIT interface sets wrong
  defaults
* added - Birthday date support for the KIT interface

Release 0.51 - 2012-02-02

* changed - reply-to handling in class.mail.php
* added - switch for email provider to enable relaying. If relaying is disabled,
  KIT will use a reply-to header instead.

Release 0.50 - 2011-12-22

* fixed - recursive dependency of kitDirList in precheck.php

Release 0.49 - 2011-12-17

* added - class.interface.php now support the check of retyped email addresses
* fixed - some typos in the language files

Release 0.48 - 2011-12-07


upgrade.php
<?php

/**
 * KeepInTouch
 *
 * @author Ralf Hertsch <ralf.hertsch@phpmanufaktur.de>
 * @link http://phpmanufaktur.de
 * @copyright 2010 - 2012
 * @license MIT License (MIT) http://www.opensource.org/licenses/MIT
 */

// include class.secure.php to protect this file and the whole CMS!
if (defined('WB_PATH')) {
  if (defined('LEPTON_VERSION'))
    include(WB_PATH.'/framework/class.secure.php');
}
else {
  $oneback = "../";
  $root = $oneback;
  $level = 1;
  while (($level < 10) && (!file_exists($root.'/framework/class.secure.php'))) {
    $root .= $oneback;
    $level += 1;
  }
  if (file_exists($root.'/framework/class.secure.php')) {
    include($root.'/framework/class.secure.php');
  }
  else {
    trigger_error(sprintf("[ <b>%s</b> ] Can't include class.secure.php!", $_SERVER['SCRIPT_NAME']), E_USER_ERROR);
  }
}
// end include class.secure.php

require_once(WB_PATH.'/modules/'.basename(dirname(__FILE__)).'/initialize.php');
require_once(WB_PATH.'/modules/'.basename(dirname(__FILE__)).'/class.mail.php');
require_once(WB_PATH.'/modules/'.basename(dirname(__FILE__)).'/class.newsletter.php');
require_once(WB_PATH.'/modules/'.basename(dirname(__FILE__)).'/class.cronjob.php');
require_once(WB_PATH.'/modules/'.basename(dirname(__FILE__)).'/class.config.php');

require_once(WB_PATH.'/framework/functions.php');

global $database;
global $admin;

$error = '';

$dbKITregister = new dbKITregister();
if (!$dbKITregister->sqlTableExists()) {
  if (!$dbKITregister->sqlCreateTable()) {
    $error .= sprintf('<p>[Upgrade] %s </p>', $dbKITregister->getError());
  }
}

$dbKITprovider = new dbKITprovider();
if (!$dbKITprovider->sqlFieldExists(dbKITprovider::field_identifier)) {
  if (!$dbKITprovider->sqlAlterTableAddField(dbKITprovider::field_identifier, "VARCHAR (50) NOT NULL DEFAULT ''", dbKITprovider::field_name)) {
    $error .= sprintf('<p>[Upgrade] %s </p>', $dbKITprovider->getError());
  }
}

$dbKITcontact = new dbKITcontact();
if (!$dbKITcontact->sqlFieldExists(dbKITcontact::field_newsletter)) {
  if (!$dbKITcontact->sqlAlterTableAddField(dbKITcontact::field_newsletter, "VARCHAR (255) NOT NULL DEFAULT ''", dbKITcontact::field_category)) {
    $error .= sprintf('<p>[Upgrade] %s </p>', $dbKITcontact->getError());
  }
}
// check field_distribution --> #0.29
if (!$dbKITcontact->sqlFieldExists(dbKITcontact::field_distribution)) {
  if (!$dbKITcontact->sqlAlterTableAddField(dbKITcontact::field_distribution, "VARCHAR(255) NOT NULL DEFAULT ''", dbKITcontact::field_newsletter)) {
     $error .= sprintf('<p>[Upgrade] %s </p>', $dbKITcontact->getError());
  }
}

// install tables for newsletter module
$dbKITnewsletterTemplates = new dbKITnewsletterTemplates();
if (!$dbKITnewsletterTemplates->sqlTableExists()) {
  if (!$dbKITnewsletterTemplates->sqlCreateTable()) {
    $error .= sprintf('<p>[Upgrade] %s</p>', $dbKITnewsletterTemplates->getError());
  }
}

$dbKITnewsletterPreview = new dbKITnewsletterPreview();
if (!$dbKITnewsletterPreview->sqlTableExists()) {
  if (!$dbKITnewsletterPreview->sqlCreateTable()) {
    $error .= sprintf('<p>[Upgrade] %s</p>', $dbKITnewsletterPreview->getError());
  }
}

$dbKITnewsletterCfg = new dbKITnewsletterCfg();
if (!$dbKITnewsletterCfg->sqlTableExists()) {
  if (!$dbKITnewsletterCfg->sqlCreateTable()) {
    $error .= sprintf('<p>[Upgrade] %s</p>', $dbKITnewsletterCfg->getError());
  }
}

$dbKITnewsletterArchive = new dbKITnewsletterArchive();
if (!$dbKITnewsletterArchive->sqlTableExists()) {
  if (!$dbKITnewsletterArchive->sqlCreateTable()) {
    $error .= sprintf('<p>[Upgrade] %s</p>', $dbKITnewsletterArchive->getError());
  }
}
// check support for distributions --> #0.29
if (!$dbKITnewsletterArchive->sqlFieldExists(dbKITnewsletterArchive::field_distributions)) {
  if (!$dbKITnewsletterArchive->sqlAlterTableAddField(dbKITnewsletterArchive::field_distributions, "VARCHAR(255) NOT NULL DEFAULT ''", dbKITnewsletterArchive::field_groups)) {
     $error .= sprintf('<p>[Upgrade] %s </p>', $dbKITnewsletterArchive->getError());
  }
}


$dbKITnewsletterProcess = new dbKITnewsletterProcess();
if (!$dbKITnewsletterProcess->sqlTableExists()) {
  if (!$dbKITnewsletterProcess->sqlCreateTable()) {
    $error .= sprintf('<p>[Upgrade] %s</p>', $dbKITnewsletterProcess->getError());
  }
}
// check process for distributions --> #0.29
if (!$dbKITnewsletterProcess->sqlFieldExists(dbKITnewsletterProcess::field_distribution_ids)) {
  if (!$dbKITnewsletterProcess->sqlAlterTableAddField(dbKITnewsletterProcess::field_distribution_ids, "VARCHAR(255) NOT NULL DEFAULT ''", dbKITnewsletterProcess::field_register_ids)) {
     $error .= sprintf('<p>[Upgrade] %s </p>', $dbKITnewsletterProcess->getError());
  }
}


$dbCronjobData = new dbCronjobData();
if (!$dbCronjobData->sqlTableExists()) {
  if (!$dbCronjobData->sqlCreateTable()) {
    $error .= sprintf('<p>[Upgrade] %s</p>', $dbCronjobData->getError());
  }
  else {
    // Blindwerte eintragen
    $datas = array(	array(dbCronjobData::field_item => dbCronjobData::item_last_call, dbCronjobData::field_value => ''),
                    array(dbCronjobData::field_item => dbCronjobData::item_last_job, dbCronjobData::field_value => ''),
                    array(dbCronjobData::field_item => dbCronjobData::item_last_nl_id, dbCronjobData::field_value => ''));
    foreach ($datas as $data) {
      if (!$dbCronjobData->sqlInsertRecord($data)) {
        $error .= sprintf('<p>[Installation] %s</p>', $dbCronjobData->getError());
      }
    }
  }
}

$dbCronjobNewsletterLog = new dbCronjobNewsletterLog();
if (!$dbCronjobNewsletterLog->sqlTableExists()) {
  if (!$dbCronjobNewsletterLog->sqlCreateTable()) {
    $error .= sprintf('<p>[Upgrade] %s</p>', $dbCronjobNewsletterLog->getError());
  }
}

$dbCronjobErrorLog = new dbCronjobErrorLog();
if (!$dbCronjobErrorLog->sqlTableExists()) {
  if (!$dbCronjobErrorLog->sqlCreateTable()) {
    $error .= sprintf('<p>[Upgrade] %s</p>', $dbCronjobErrorLog->getError());
  }
}

/**
 * BUGFIX correct a problem of KIT < 0.34 with duplicate active entries for the same e-mail address within dbKITregister
 */
$SQL = sprintf("SELECT %s, %s, COUNT(*) AS cnt FROM %s WHERE %s='%s' GROUP BY %s HAVING cnt>1",
                dbKITregister::field_contact_id,
                dbKITregister::field_id,
                $dbKITregister->getTableName(),
                dbKITregister::field_status,
                dbKITregister::status_active,
                dbKITregister::field_email);
$registers = array();
if (!$dbKITregister->sqlExec($SQL, $registers)) {
  $error .= sprintf('<p>[BUGFIX] %s</p>', $dbKITregister->getError());
}
else {
  foreach ($registers as $register) {
    $SQL = sprintf( "SELECT %s FROM %s WHERE %s='%s' AND %s='%s'",
                    dbKITcontact::field_id,
                    $dbKITcontact->getTableName(),
                    dbKITcontact::field_id,
                    $register[dbKITregister::field_contact_id],
                    dbKITcontact::field_status,
                    dbKITcontact::status_deleted);
    $contacts = array();
    if (!$dbKITcontact->sqlExec($SQL, $contacts)) {
      $error .= sprintf('<p>[BUGFIX] %s</p>', __METHOD__, __LINE__, $dbKITcontact->getError());
    }
    else {
      if (count($contacts) > 0) {
        $where = array(dbKITregister::field_id => $register[dbKITregister::field_id]);
        $data = array(
          dbKITregister::field_status 			=> dbKITregister::status_deleted,
          dbKITregister::field_update_by		=> 'UPDATE FIXUP',
          dbKITregister::field_update_when	=> date('Y-m-d H:i:s')
        );
        if (!$dbKITregister->sqlUpdateRecord($data, $where)) {
          $error .= sprintf('<p>[BUGFIX] %s</p>', $dbKITregister->getError());
        }
      }
    }
  }
}

/**
 * Release 0.43
 * Remove /dialogs, /droplets, kit.php, class.request.php and class.repsonse.php and use kitForms instead
 */

// remove mod_kit_newsletter_links
$SQL = sprintf("DROP TABLE IF EXISTS %smod_kit_newsletter_links", TABLE_PREFIX);
$database->query($SQL);
if ($database->is_error()) {
  $error .= sprintf('<p>[DROP TABLE mod_kit_newsletter_links] %s</p>', $database->get_error());
}

// remove Droplet kit_newsletter
$SQL = sprintf("DELETE FROM %smod_droplets WHERE name='kit_newsletter'", TABLE_PREFIX);
$database->query($SQL);
if ($database->is_error()) {
  $error .= sprintf('<p>[DELETE DROPLET kit_newsletter] %s</p>', $database->get_error());
}

// delete no longer needed entries from mod_kit_config
$SQL = sprintf("DELETE FROM %smod_kit_config WHERE cfg_name IN ('cfgLicenseKey','cfgKITResponsePage','cfgUseCaptcha','cfgUseCustomFiles','cfgRegisterDlgNL','cfgRegisterDlgACC','cfgRegisterDlgUSUB','cfgKITRequestLink','cfgMaxInvalidLogin','cfgMinPwdLen')", TABLE_PREFIX);
$database->query($SQL);
if ($database->is_error()) {
  $error .= sprintf('<p>[DELETE ENTRIES FROM mod_kit_config] %s</p>', $database->get_error());
}

// delete no longer needed files and directories
$delete_array = array('kit.php', 'class.newsletter.link.php', 'class.response.php', 'class.request.php', 'droplets', 'class.droplets.php', 'class.dialogs.php', 'dialogs', 'frontend.css');
foreach ($delete_array as $item) {
  if (file_exists(WB_PATH.'/modules/kit/'.$item)) {
    if (!rm_full_dir(WB_PATH.'/modules/kit/'.$item)) {
      $error .= sprintf('<p>[DELETE FILES/DIRECTORIES] Can\'t delete /modules/kit/%s</p>', $item);
    }
  }
}

/**
 * Release 0.46
 */
// change field lenght for values from 80 chars to TEXT
if (!$dbCronjobData->sqlAlterTableChangeField(dbCronjobData::field_value, dbCronjobData::field_value, "TEXT NOT NULL DEFAULT ''")) {
    $error .= sprintf('[<p>[ALTER TABLE mod_kit_cronjob_data] %s</p>', $dbCronjobData->getError());
}

/**
 * Release 0.51
 */
// add field 'relaying' to email provider
if (!$dbKITprovider->sqlFieldExists(dbKITprovider::field_relaying)) {
    if (!$dbKITprovider->sqlAlterTableAddField(dbKITprovider::field_relaying, "TINYINT NOT NULL DEFAULT '0'", dbKITprovider::field_smtp_pass)) {
        $error .= sprintf('<p>[ALTER TABLE mod_kit_provider] %s</p>', $dbKITprovider->getError());
    }
}

/**
 * Release 0.54
 */
// new table mod_kit_languages
$dbKITlanguages = new dbKITlanguages();
if (!$dbKITlanguages->sqlTableExists()) {
  if (!$dbKITlanguages->initTables()) {
    $error .= sprintf('<p>[CREATE TABLE mod_kit_languages] %s</p>', $dbKITlanguages->getError());
  }
}
// add field 'contact_language' to contact db
if (!$dbKITcontact->sqlFieldExists(dbKITcontact::field_contact_language)) {
  if (!$dbKITcontact->sqlAlterTableAddField(dbKITcontact::field_contact_language, "VARCHAR(2) NOT NULL DEFAULT 'en'", dbKITcontact::field_contact_note)) {
    $error .= sprintf('<p>[ALTER TABLE mod_kit_contact] %s</p>', $dbKITcontact->getError());
  }
}

/**
 * Release 0.56
 */
$dbKITcfg = new dbKITcfg();
// remove cfgGoogleMapsAPIkey
$where = array(dbKITcfg::field_name => 'cfgGoogleMapsAPIkey');
if (!$dbKITcfg->sqlDeleteRecord($where)) {
  $error .= sprintf('<p>[UPDATE CONFIG TABLE mod_kit_config] %s</p>', $dbKITcfg->getError());
}

/**
 * Release 0.62
 */
$where = array(dbKITcfg::field_name => 'cfgSortContactList');
if (!$dbKITcfg->sqlDeleteRecord($where)) {
  $error .= sprintf('<p>[UPDATE CONFIG TABLE mod_kit_config] %s</p>', $dbKITcfg->getError());
}
$where = array(dbKITcfg::field_name => 'cfgLimitContactList');
if (!$dbKITcfg->sqlDeleteRecord($where)) {
  $error .= sprintf('<p>[UPDATE CONFIG TABLE mod_kit_config] %s</p>', $dbKITcfg->getError());
}

/**
 * Release 0.63
 */
// create the kit_link table
$SQL = "CREATE TABLE IF NOT EXISTS `".TABLE_PREFIX."mod_kit_links` ( ".
    "`id` INT(11) NOT NULL AUTO_INCREMENT, ".
    "`url` TEXT NOT NULL, ".
    "`guid` VARCHAR(128) NOT NULL DEFAULT '', ".
    "`type` ENUM('DOWNLOAD','UPLOAD','UNDEFINED') NOT NULL DEFAULT 'UNDEFINED', ".
    "`option` ENUM('THROW-AWAY','PERMANENT') NOT NULL DEFAULT 'THROW-AWAY', ".
    "`status` ENUM('ACTIVE','LOCKED','DELETED') NOT NULL DEFAULT 'ACTIVE', ".
    "`file_url` TEXT NOT NULL, ".
    "`count` INT(11) NOT NULL DEFAULT '0', ".
    "`last_call` DATETIME NOT NULL DEFAULT '0000-00-00 00:00:00', ".
    "`kit_id` INT(11) NOT NULL DEFAULT '-1', ".
    "`timestamp` TIMESTAMP, ".
    "PRIMARY KEY (`id`), ".
    "KEY (`guid`,`kit_id`)".
    ") ENGINE=MyIsam AUTO_INCREMENT=1 DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci";
$database->query($SQL);
if ($database->is_error()) {
  $error .= sprintf('[INSTALLATION kit_links] %s', $database->get_error());
}

/**
 * Release 0.67
 */

@unlink(WB_PATH.'/modules/kit/htt/backend.config.import.htt');
@unlink(WB_PATH.'/modules/kit/htt/backend.config.import.massmail.htt');
@unlink(WB_PATH.'/modules/kit/htt/backend.config.import.massmail.tr.htt');
@unlink(WB_PATH.'/modules/kit/htt/backend.import.step.1.htt');
@unlink(WB_PATH.'/modules/kit/htt/backend.import.step.1.tr.htt');
@unlink(WB_PATH.'/modules/kit/htt/backend.import.step.2.htt');
@unlink(WB_PATH.'/modules/kit/htt/backend.import.step.2.th.htt');
@unlink(WB_PATH.'/modules/kit/htt/backend.import.step.2.tr.htt');
@unlink(WB_PATH.'/modules/kit/htt/backend.newsletter.config.csv.htt');
@unlink(WB_PATH.'/modules/kit/htt/backend.start.register.dlg.htt');
@unlink(WB_PATH.'/modules/kit/htt/backend.start.register.dlg.tr.htt');
@unlink(WB_PATH.'/modules/kit/htt/newsletter.sample.wbugb.htt');
@unlink(WB_PATH.'/modules/kit/htt/frontend.error.htt');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/addresses.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/categories.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/company.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/contact_since.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/distribution.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/email.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/information.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/internet.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/map.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/newsletter.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/notes.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/person.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/include/contact/phone.lte');
@unlink(WB_PATH.'/modules/kit/templates/backend/backend.contact.lte');

/**
 * Release 0.69
 */

$dbKITaddress = new dbKITcontactAddress();
// additional fields for the contact address
if (!$dbKITaddress->sqlFieldExists(dbKITcontactAddress::field_extra)) {
  if (!$dbKITaddress->sqlAlterTableAddField(dbKITcontactAddress::field_extra, "VARCHAR(255) NOT NULL DEFAULT ''", dbKITcontactAddress::field_city)) {
    $error .= sprintf('<p>[ALTER TABLE mod_kit_contact] %s</p>', $dbKITaddress->getError());
  }
}
if (!$dbKITaddress->sqlFieldExists(dbKITcontactAddress::field_region)) {
  if (!$dbKITaddress->sqlAlterTableAddField(dbKITcontactAddress::field_region, "VARCHAR(255) NOT NULL DEFAULT ''", dbKITcontactAddress::field_extra)) {
    $error .= sprintf('<p>[ALTER TABLE mod_kit_contact] %s</p>', $dbKITaddress->getError());
  }
}

// Prompt Errors
if (!empty($error)) {
  $admin->print_error($error);
}



?>